datastore.go

```diff
@@ -29,6 +29,8 @@ type Datastore struct {
 	gcDiscardRatio float64
 	gcSleep        time.Duration
 	gcInterval     time.Duration
+
+	syncWrites bool
```

Comment:
@Stebalien should we make async writes the default in the datastore, or just the default in IPFS?

Reply:
Let's leave the datastores as-is and make it the default in go-ipfs. Making it the default for _everyone_ is a large leap.